real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/localizers/HighTrustGPSLocalizer.java

```diff
@@ -39,51 +39,65 @@ public HighTrustGPSLocalizer(){
     	buggyFrameGpsX = 0.0;
     	buggyFrameGpsY = 0.0;
         buggyFrameRotZ = 0.0;
+        buggySteeringAngle = 0.0;// wheel direction in buggy frame
         lastEncoderReading = 0.0;
         posePub = new Publisher(NodeChannel.POSE.getMsgPath());
-        mostRecentUpdate = new Date();
-
 
+        //Read all the steering messages we're getting and adjust our stored steering angle accordingly
+        new Subscriber("htGpsLoc", NodeChannel.STEERING.getMsgPath(), new MessageListener() {			
+			@Override
+			public void actionPerformed(String topicName, Message m) {
+				SteeringMeasurement steerM = (SteeringMeasurement)m;
+				// TODO Auto-generated method stub
+	            buggySteeringAngle  = steerM.getAngle();
+			}
+		});
+        
         //Initialize subscriber to GPS measurements
         new Subscriber("htGpsLoc", NodeChannel.GPS.getMsgPath(), new MessageListener() {
             @Override
             public void actionPerformed(String topicName, Message m) {
                 GpsMeasurement newGPSData = (GpsMeasurement)m;
                 synchronized (this) {
-                    long dt = newGPSData.getTimestamp().getTime() - mostRecentUpdate.getTime();
-                    if(dt > 0.0){
-                          // Get the delta latitude and longitude, use that to figure out how far we've travelled
-
-//               double oldGPSX = buggyFrameGpsX;
-//                 double oldGPSY = buggyFrameGpsY;
-                	buggyFrameGpsY = newGPSData.getLatitude();
-                	buggyFrameGpsX = newGPSData.getLongitude();
-    //            double dLat = buggyFrameGpsY - oldGPSY;
-    //            double dLon = buggyFrameGpsX - oldGPSX;
-                
-    //            double oldRotZ = buggyFrameRotZ;
+                // Get the delta latitude and longitude, use that to figure out how far we've travelled
+               buggyFrameGpsY = newGPSData.getLatitude();
+               buggyFrameGpsX = newGPSData.getLongitude();
+               double dLat = buggyFrameGpsY - oldGPSY;
+               double dLon = buggyFrameGpsX - oldGPSX;
+               oldGPSX = buggyFrameGpsX;
+               oldGPSY = buggyFrameGpsY;
 
                 // take the arctangent in order to get the heading (in degrees)
-         //     buggyFrameRotZ = Math.toDegrees(Math.atan2(dLat, dLon));
+                buggyFrameRotZ = Math.toDegrees(Math.atan2(LocalizerUtil.convertLatToMeters(dLat), LocalizerUtil.convertLonToMeters(dLon)));
 
                         publishUpdate();
-                        mostRecentUpdate = newGPSData.getTimestamp();
                     }
                 }
-            }
         });
 
-        
+ /*
         new Subscriber("HighTrustGpsLoc",NodeChannel.IMU_ANG_POS.getMsgPath(), ((topicName, m) -> {
```

Comment:
I think what happened is that we were trying to do everything through matrices, but then we commented it out in favor of the kalman filter since the hillcrest died